### PR TITLE
Enable Shippable

### PR DESCRIPTION
**Pull request by odyssey4me**
##### SUMMARY

Basic Shippable configuration, based on community.general.

Fixes: #1

##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by gundalow** [comment]
Ah, I hadn't enabled Shippable before.

File: tests/utils/shippable/shippable.sh [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+virtualenv --python /usr/bin/python3.7 ~/ansible-venv
+set +ux
+. ~/ansible-venv/bin/activate
+set -ux
+
+#pip install ansible==2.9.0 --disable-pip-version-check
+pip install git+https://github.com/ansible/ansible.git@temp-2.10-devel --disable-pip-version-check
+
+COLLECTION_DIR="${HOME}/.ansible/ansible_collections/"
+TEST_DIR="${COLLECTION_DIR}/community/rabbitmq"
```
Comment (gundalow) [inline tests/utils/shippable/shippable.sh]:
```suggestion
TEST_DIR="${COLLECTION_DIR}/community/libvirt"
```

**Comment by odyssey4me** [comment]
@gundalow Yeah, I figured as much. Thanks!

File: tests/utils/shippable/shippable.sh [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    export UNSTABLE=""
+fi
+
+virtualenv --python /usr/bin/python3.7 ~/ansible-venv
+set +ux
+. ~/ansible-venv/bin/activate
+set -ux
+
+#pip install ansible==2.9.0 --disable-pip-version-check
+pip install git+https://github.com/ansible/ansible.git@temp-2.10-devel --disable-pip-version-check
```
Comment (gundalow) [inline tests/utils/shippable/shippable.sh]: The `devel` branch has been updated. Also downloading the tar.gz file is a lot faster than git clone.

```suggestion
pip install https://github.com/ansible/ansible/archive/devel.tar.gz --disable-pip-version-check
```

File: tests/utils/shippable/shippable.sh [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+# STAR: HACK install dependencies
+(
+#ansible.windows doesn't install from Galaxy
+mkdir /tmp/collection_deps
+git clone https://github.com/ansible-collections/ansible.windows.git /tmp/collection_deps/ansible.windows
+cd /tmp/collection_deps/ansible.windows
+ansible-galaxy collection build
+ansible-galaxy collection install /tmp/collection_deps/ansible.windows/ansible-windows* -p "${COLLECTION_DIR}"
+)
+
```
Comment (gundalow) [inline tests/utils/shippable/shippable.sh]: You don't require windows
```suggestion
```

File: tests/utils/shippable/shippable.sh [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+#ansible-galaxy collection install ansible.windows -p "${COLLECTION_DIR}"
```
Comment (gundalow) [inline tests/utils/shippable/shippable.sh]: Most of the others can do, though need to check for any Python or integration/unit tests imports

File: tests/utils/shippable/shippable.sh (hunk 3 above)
Comment (odyssey4me) [reply tests/utils/shippable/shippable.sh]: Done.

File: tests/utils/shippable/shippable.sh (hunk 4 above)
Comment (odyssey4me) [reply tests/utils/shippable/shippable.sh]: Are these installs really required? It doesn't seem to me like these are used anywhere in this collection.

File: tests/utils/shippable/shippable.sh (hunk 1 above)
Comment (odyssey4me) [reply tests/utils/shippable/shippable.sh]: Done.

File: tests/utils/shippable/shippable.sh (hunk 2 above)
Comment (odyssey4me) [reply tests/utils/shippable/shippable.sh]: Done.

**Comment by odyssey4me** [comment]
The first issue to sort out for sanity tests is:

```
01:10 Run command: importer.py
01:10 ERROR: Command "importer.py" returned exit status 1.
01:10 >>> Standard Error
01:10 Traceback (most recent call last):
01:10   File "/root/ansible/ansible_collections/community/libvirt/tests/output/.tmp/sanity/import/minimal-py26/bin/importer.py", line 447, in <module>
01:10     main()
01:10   File "/root/ansible/ansible_collections/community/libvirt/tests/output/.tmp/sanity/import/minimal-py26/bin/importer.py", line 51, in main
01:10     from ansible.utils.collection_loader import AnsibleCollectionLoader
01:10   File "/root/ansible/lib/ansible/utils/collection_loader.py", line 15, in <module>
01:10     from ansible import constants as C
01:10   File "/root/ansible/lib/ansible/constants.py", line 12, in <module>
01:10     from jinja2 import Template
01:10 ImportError: No module named jinja2
```

It looks like the requirements for ansible-test are inside a docker file. Is there a way to add to those somewhere?

**Comment by gundalow** [comment]
Hum, seems to be a regression in devel over the past day or so https://github.com/ansible/ansible/issues/68530

If you go back to using the temp branch you should get further.

**Comment by odyssey4me** [comment]
In the unit test, the following failure is happening:

```
00:42 ImportError while loading conftest '/root/ansible/ansible_collections/community/libvirt/tests/unit/modules/cloud/misc/virt_net/conftest.py'.
00:42 ../../../lib/ansible/utils/collection_loader.py:129: in load_module
00:42     mod = self._find_module(fullname, None, load=True)[1]
00:42 ../../../lib/ansible/utils/collection_loader.py:269: in _find_module
00:42     exec(code_object, newmod.__dict__)
00:42 tests/unit/modules/cloud/misc/virt_net/conftest.py:9: in <module>
00:42     from ansible_collections.community.general.tests.unit.compat import mock
00:42 E   ImportError: No module named general.tests.unit.compat
```

**Comment by odyssey4me** [comment]
@gundalow OK, that's all but the unit tests sorted out. I'm sure that can be solved easily enough. It's likely a pathing or lack of content issue. I'll replicate these fixes over to the rabbitmq PR.

**Comment by gundalow** [comment]
Ansible-test doesn't know how to import other collections for unit tests. Maybe just copy the directory from https://github.com/ansible-collections/ansible.windows/tree/master/tests/unit/ compat & mock, and update the python import statements so they don't reference community.general

**Comment by odyssey4me** [comment]
@gundalow Thanks for the pointers - we now have test parity with what was being tested in ansible/ansible. It's not much, but it's a start!

**Pull request merged**